source/patterns/@aws-solutions-constructs/aws-s3-sns/test/test.s3-sns.test.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { Stack } from "aws-cdk-lib";
+import { S3ToSns } from "../lib";
+import '@aws-cdk/assert/jest';
+import * as defaults from '@aws-solutions-constructs/core';
+import * as kms from 'aws-cdk-lib/aws-kms';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+
+test('All get methods return non-null objects', () => {
+  const stack = new Stack();
+  const app = new S3ToSns(stack, 'test-s3-sns', {
+
+  });
+  expect(app.snsTopic !== null);
+  expect(app.s3Bucket !== null);
+  expect(app.s3LoggingBucket !== null);
+  expect(app.encryptionKey !== null);
+  expect(app.s3BucketInterface !== null);
+});
+
+test('deployment works with existing bucket', () => {
+  const stack = new Stack();
+  const [ existingBucketObj ] = defaults.buildS3Bucket(stack, {});
+  new S3ToSns(stack, 'test-s3-sns', {
+    existingBucketObj
+  });
+
+  expect(stack).toHaveResource("Custom::S3BucketNotifications", {
+    NotificationConfiguration: {
+      TopicConfigurations: [
+        {
+          Events: ['s3:ObjectCreated:*'],
+          TopicArn: {
+            Ref: "tests3snsSnsTopicF02F6BD0"
+          }
+        }
+      ]
+    }
+  });
+});
+
+test('deployment works with existing topic and key', () => {
+  const stack = new Stack();
+  const cmk = defaults.buildEncryptionKey(stack);
+  const [ existingTopicObj ] = defaults.buildTopic(stack, {
+    encryptionKey: cmk
+  });
+  new S3ToSns(stack, 'test-s3-sns', {
+    existingTopicObj,
+    existingTopicEncryptionKey: cmk
+  });
+
+  expect(stack).toHaveResource("Custom::S3BucketNotifications", {
+    NotificationConfiguration: {
+      TopicConfigurations: [
+        {
+          Events: ['s3:ObjectCreated:*'],
+          TopicArn: {
+            Ref: "SnsTopic2C1570A4"
+          }
+        }
+      ]
+    }
+  });
+
+  expect(stack).toHaveResource("AWS::SNS::Topic", {
```

Comment:
Again - give the topic a unique attrib (eg - a specific name) and confirm that attrib is there. Drop the brittle generated ID check and use toHaveResourceLike that will allow you to check for just the specific topic name.